cloudify_hostpool/rest/backend.py

```diff
@@ -125,3 +128,16 @@ def _is_alive(host):
         address, port = host['host'], host['port']
         results = scan.scan([(address, port)])
         return results[address, port]
+
+    @staticmethod
+    def _load_keyfile(host):
+        if not host['auth']:
```

Comment:
You mean to cache key contents in memory upon configuration? Holding them in the database would be an antipattern.